### PR TITLE
Post install delegation checker

### PR DESCRIPTION
**Pull request by bitscuit**
Implements a new controller `PostDelegationChecker` to check whether delegation to another cert-manager is needed after this operator has already been installed.

Situation that this covers is when this operator is already installed and the operands are already created. Then user installs another cert-manager onto the cluster.

## How to test

1. See how to test with by running bundle https://github.com/IBM/ibm-cert-manager-operator/blob/release-ltsr/CONTRIBUTING.md#testing-bundle-with-olm
2. After running the bundle, operator pod should be running on the cluster
3. Create the operands by running `oc apply -f config/samples/operator_v1_certmanagerconfig.yaml`
4. Verify that operands are created
5. Install another cert-manager, e.g. Red Hat cert-manager
6. Verify that IBM operands are deleted, and RH cert-manager is running
7. Verify that you can create a self-signed Issuer and CA certificate

### CODE REVIEW

File: controllers/operator/postdelegationchecker_controller.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	}
+	configmap.Data[OPERAND_TOGGLE] = "false"
+
+	if err := r.Client.Update(ctx, configmap); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Successfully configured ibm-cpp-config")
+
+	logger.Info("Deleting default " + CERTIFICATE_API + " instance")
```
Comment (Daniel-Fan) [inline controllers/operator/postdelegationchecker_controller.go:140]: The info msg is a little bit confused.
```
const CERTMANAGER_API = "CertManager.v1alpha1.operator.ibm.com"
logger.Info("Deleting default " + CERTMANAGER_API + " instance")
```

**Review by Daniel-Fan — COMMENTED** [review]
Minor issue when I test with `make bundle`
```
✗ make bundle                                                                                 
/root/ibm-cert-manager-operator/bin/controller-gen rbac:roleName=operator crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
common/scripts/patch-metadata.sh /root/ibm-cert-manager-operator/bin/yq
common/scripts/delete_v1.sh
 cert-manager.io_certificates.yaml is removed
 cert-manager.io_issuers.yaml is removed
common/scripts/patch-csv.sh v3.25.7 3.25.8
operator-sdk generate kustomize manifests -q
WARN[0000] Skipping definitions parsing for API {}: Go type not found 
WARN[0000] Skipping definitions parsing for API {}: Go type not found 
cd config/manager && ./bin/kustomize edit set image controller=ibm-cert-manager-operator
bash: line 1: ./bin/kustomize: No such file or directory
make: *** [Makefile:257: bundle] Error 127
```
We could update [this line](https://github.com/IBM/ibm-cert-manager-operator/blob/1818af9451833894e01d1ac7fd8aab68591a971a/Makefile#L257) to
```
cd config/manager && ../../$(KUSTOMIZE) edit set image controller=$(IMG)

**Review by Daniel-Fan — COMMENTED** [review]
/lgtm

File: controllers/operator/postdelegationchecker_controller.go (hunk 1 above)
Comment (bitscuit) [reply controllers/operator/postdelegationchecker_controller.go:140]: Oh, yea, not sure why I put certificate api there. I will change in another PR, and hold it for next release.

**Pull request merged**